Src/FluentAssertions/Xml/XmlReaderValidator.cs

```diff
@@ -77,7 +101,15 @@ private ValidationResult Validate()
                             return validationResult;
                         }
 
+                        // starting new element, add local name to location stack
+                        // to build XPath info
                         locationStack.Push(subjectReader.LocalName);
+
+                        // building key according XPath to count repeating element names
+                        var locationKey = "/" + string.Join("/", locationStack.Reverse());
```

Comment:
I don't get this. Aren't you building the entire path expression? How can that be used as a key?

Reply:
Given the xml `<a><b /><b /></a>`
* `<a>`: 
  * push `a` on the stack
  * update the dictionary with `/a` = 1
* `<b />`
  * push `b` on the stack
  * update the dictionary with `/a/b` = 1 <-- first time we see `<b>` on the `/a` xpath
  * pop `b` from the stack
* `<b />`
  * push `b` on the stack
  * update the dictionary with `/a/b` = 2 <-- second time we see `<b>` on the `/a` xpath
  * pop `b` from the stack
* `</a>`
  * pop `a` from the stack